src/cmd/manage_nsfs.js

```diff
@@ -242,49 +239,46 @@ async function delete_bucket(data, force) {
         }
         await native_fs_utils.folder_delete(bucket_temp_dir_path, fs_context_fs_backend, true);
         await config_fs.delete_bucket_config_file(data.name);
-        write_stdout_response(ManageCLIResponse.BucketDeleted, '', { bucket: data.name });
+        return { code: ManageCLIResponse.BucketDeleted, detail: '', event_arg: { bucket: data.name } };
     } catch (err) {
         if (err.code === 'ENOENT') throw_cli_error(ManageCLIError.NoSuchBucket, data.name);
         throw err;
     }
 }
 
-async function manage_bucket_operations(action, data, user_input) {
+/**
+ * bucket_management does the following - 
+ * 1. fetches the bucket data if this is not a list operation
+ * 2. validates bucket args - TODO - we should split it to validate_bucket_args 
+ * and validations of the merged (user_input + existing bucket config) bucket
+ * 3. call bucket operation based on the action argument
+ * 4. write output to stdout
+ * @param {String} action 
```

Comment:
If you want we can improve the JSDoc for action (not only here):
```suggestion
 * @param {(''add''|'update'|'delete'|'status'|'list')} action
```